dragonfly/implementations/roach_daq_run_interface.py

```diff
@@ -0,0 +1,338 @@
+
+        #check psyllid is responding by requesting status
+        self.status_value = self.provider.cmd(self.psyllid_interface, 'request_status', payload = self.payload_channel)['values'][0]
+        if self.status_value == None:
+            logger.error('Psyllid is not responding')
+            raise core.exceptions.DriplineGenericDAQError('Psyllid is not responding')        
+
+        #check channel match
+        active_channels = self.provider.get(self.psyllid_interface + '.active_channels')
+        logger.info(active_channels)
+        if self.channel_id in active_channels==False:
+            logger.error('The Psyllid and ROACH2 channel interfaces do not match')
+            raise core.exceptions.DriplineGenericDAQError('The Psyllid and ROACH channel interfaces do not match')
+
+
+    def _do_checks(self):
+        if self._run_time ==0:
+            raise core.exceptions.DriplineValueError('run time is zero')
+        #elif self._run_time >= self._max_duration:
+        #    raise core.exceptions.DriplineValueError('run time exceeds max duration')
+
+        #checking psyllid
+        if self.is_running==True:
+            raise core.exceptions.DriplineGenericDAQError('Psyllid is already running')
+
+        self._check_psyllid_instance()
+
+        if self.status_value!=4:
+            raise core.exceptions.DriplineGenericDAQError('Psyllid DAQ is not activated')
+
+        #checking roach
+        if self._check_roach2_status() == False:
+            raise core.exceptions.DriplineGenericDAQError('ROACH2 is not ready')
+        blocked_channels = self.provider.get(self.daq_target+ '.blocked_channels')
+        logger.info(blocked_channels)
+        if self.channel_id in blocked_channels: 
+            raise core.exceptions.DriplineGenericDAQError('Channel is blocked')
+
+        #check frequency matches
+        roach_freqs = self._get_roach_central_freqs()
+        psyllid_freq = self._get_psyllid_central_freq()
+        if roach_freqs[self.channel_id]!=psyllid_freq:
+            raise core.exceptions.DriplineGenericDAQError('Frequency mismatch')
+
+        return "checks successful"
+
+
+    def determine_RF_ROI(self):
+        logger.info('trying to determine roi')
+
+        self._run_meta['RF_HF_MIXING'] = float(self._hf_lo_freq)
+        logger.debug('RF High stage mixing: {}'.format(self._run_meta['RF_HF_MIXING']))
+
+        logger.info('Getting central frequency from ROACH2')
+        cfs = self._get_roach_central_freqs()
+        cf = cfs[self.channel_id]
+        logger.info('Central frequency is: {}'.format(cf))
+
+        self._run_meta['RF_ROI_MIN'] = float(cf-50e6) + float(self._hf_lo_freq)
+        logger.debug('RF Min: {}'.format(self._run_meta['RF_ROI_MIN']))
+
+        self._run_meta['RF_ROI_MAX'] = float(cf+50e6) + float(self._hf_lo_freq)
+        logger.debug('RF Max: {}'.format(self._run_meta['RF_ROI_MAX']))
+
+
+
+    def _start_data_taking(self, directory, filename):
```

Comment:
This is probably historic from before Mathieu started working on it. The mantis interface was done first and did not use timeouts. You'd add run requests of fixed length to the run queue then poll the queue to see when it was done/empty. Adding support for the RSA required a change because it need to be explicitly told to start/stop.

The better solution (discussed on slack) would be for me to clean up the scheduler class and for that to be used rather than dealing with pika's connection object timeouts in other classes.